primitives/test-utils/src/lib.rs

```diff
@@ -118,11 +123,53 @@ pub fn header_id<H: HeaderT>(index: u8) -> (H::Hash, H::Number) {
 	(test_header::<H>(index.into()).hash(), index.into())
 }
 
-/// Get the identity of a test account given an ED25519 Public key.
-pub fn extract_keyring(id: &AuthorityId) -> Ed25519Keyring {
-	let mut raw_public = [0; 32];
-	raw_public.copy_from_slice(id.as_ref());
-	Ed25519Keyring::from_raw_public(raw_public).unwrap()
+/// Set of test accounts.
+#[derive(RuntimeDebug, Clone, Copy)]
+pub enum Keyring {
+	Alice,
+	Bob,
+	Charlie,
+	Dave,
+	Eve,
+	Ferdie,
+}
+
+impl Keyring {
+	pub fn public(&self) -> PublicKey {
+		(&self.secret()).into()
+	}
+
+	pub fn secret(&self) -> SecretKey {
+		SecretKey::from_bytes(&[*self as u8; 32]).expect("A static array of the correct length is a known good.")
+	}
+
+	pub fn pair(self) -> Keypair {
+		let mut pair: [u8; 64] = [0; 64];
+
+		let secret = self.secret();
+		pair[..32].copy_from_slice(&secret.to_bytes());
+
+		let public = self.public();
+		pair[32..].copy_from_slice(&public.to_bytes());
+
+		Keypair::from_bytes(&pair).expect("We expect the SecretKey to be good, so this must also be good.")
+	}
+
+	pub fn sign(self, msg: &[u8]) -> Signature {
+		self.pair().sign(msg)
+	}
+}
+
+impl From<Keyring> for AuthorityId {
+	fn from(k: Keyring) -> Self {
+		AuthorityId::from_slice(&k.public().to_bytes())
+	}
+}
+
+impl From<&Keyring> for AuthorityId {
+	fn from(k: &Keyring) -> Self {
+		AuthorityId::from_slice(&k.public().to_bytes())
```

Comment:
```suggestion
		(*k).into()
```
In case this is actually needed (See other comment).